aiida_quantumespresso/calculations/pp.py

```diff
@@ -1,27 +1,240 @@
+            for namelist_name in namelists_toprint:
+                infile.write(u'&{0}\n'.format(namelist_name))
+                # namelist content; set to {} if not present, so that we leave an empty namelist
+                namelist = parameters.pop(namelist_name, {})
+                for key, value in sorted(six.iteritems(namelist)):
+                    infile.write(convert_input_to_namelist_entry(key, value))
+                infile.write(u'/\n')
+
+
+        # Check for specified namelists that are not expected
+        if parameters:
+            raise exceptions.InputValidationError(
+                'The following namelists are specified in parameters, but are '
+                'not valid namelists for the current type of calculation: '
+                '{}'.format(','.join(list(parameters.keys()))))
+
+        remote_copy_list = []
+        local_copy_list = []
+
+        # copy remote output dir, if specified
+        parent_calc_folder = self.inputs.get('parent_folder', None)
+        if parent_calc_folder is not None:
+            if isinstance(parent_calc_folder, orm.RemoteData):
+                parent_calc_out_subfolder = settings.pop('PARENT_CALC_OUT_SUBFOLDER', self._INPUT_SUBFOLDER)
+                remote_copy_list.append((
+                    parent_calc_folder.computer.uuid,
+                    os.path.join(parent_calc_folder.get_remote_path(), parent_calc_out_subfolder),
+                    self._OUTPUT_SUBFOLDER
+                ))
+            elif isinstance(parent_calc_folder, orm.FolderData):
+                for filename in parent_calc_folder.list_object_names():
+                    local_copy_list.append((
+                        parent_calc_folder.uuid,
+                        filename,
+                        os.path.join(self._OUTPUT_SUBFOLDER, filename)
+                    ))
+            elif isinstance(parent_calc_folder, orm.SinglefileData):
+                single_file = parent_calc_folder
+                local_copy_list.append(
+                    (single_file.uuid, single_file.filename, single_file.filename)
+                )
+
+        codeinfo = datastructures.CodeInfo()
+        codeinfo.cmdline_params = settings.pop('CMDLINE', [])
+        codeinfo.stdin_name = self.inputs.metadata.options.input_filename
+        codeinfo.stdout_name = self.inputs.metadata.options.output_filename
+        codeinfo.code_uuid = self.inputs.code.uuid
+
+        calcinfo = datastructures.CalcInfo()
+        calcinfo.uuid = str(self.uuid)
+        calcinfo.codes_info = [codeinfo]
+        calcinfo.local_copy_list = local_copy_list
+        calcinfo.remote_copy_list = remote_copy_list
+
+        # Retrieve by default the output file and the xml file
+        calcinfo.retrieve_list = []
+        calcinfo.retrieve_list.append(self.inputs.metadata.options.output_filename)
+        settings_retrieve_list = settings.pop('ADDITIONAL_RETRIEVE_LIST', [])
+        calcinfo.retrieve_list += settings_retrieve_list
```

Comment:
Does this really make sense for pp.x or is this just copied from another calcjob implementation. If not necessary I prefer to remove it, and then also the `settings` input node as this is the only usage as far as I can tell

Reply:
Agreed.

Reply:
Actually the only part of setting that really makes sense is that the output subfolder in the parent may be called something other than "./out"

Reply:
AFAICT the output subfolder is not user-definable, it's hard-coded in the `cls._OUTPUT_SUBFOLDER` variable.